### PR TITLE
Add CI wheel build for MacOS

### PR DESCRIPTION
**Pull request by xmatthias**
Adding support for MacOS builds, for both x64 and arm64 builds.
Only the x64 are being tested - and I'd be grateful if someone can step up and try to install the arm64 binary on a M1 system.

A recent CI run for this can be found [here](https://github.com/xmatthias/PyTables/runs/4535307675?check_suite_focus=true).

This is based mostly on the great work from @amotl in #872

---

This is currently using the hdf5 (and blosc) library from brew (as well as some other libraries necessary for building).
While this might for sure be seen as controversial, i don't see this as worse than using libraries from conda (which pytables does and did for Windows CI already).

I'm also working on a "build hdf5 from source" PR - which will build hdf5 from source on linux/macos Systems - but the brew dependency will probably remain there for some "basic" libraries we don't really want to build.

I'm not sure if it would be better to use the builtin / vendored blosc (I'm not sure on reasons for vendoring that library - but discovered that it'll be built as part of the wheel building process if the library is not there).

### CODE REVIEW

File: .github/workflows/wheels.yml [hunk 1]
```diff
@@ -35,19 +34,32 @@ jobs:
...
+      - name: Install prerequisites for macOS
+        if: contains(matrix.os, 'macos')
+        env:
+          # Best compatibility, even with older releases of macOS.
+          MACOSX_DEPLOYMENT_TARGET: "10.9"
+        run: |
+          brew install --build-from-source --no-binaries --force szip
+          brew reinstall --build-from-source --no-binaries --force c-blosc bzip2 hdf5 lz4 lzo snappy zstd zlib
+          brew link --overwrite --force bzip2 zlib
+
```
Comment (avalentino) [inline .github/workflows/wheels.yml:46]: Personally I would feel more comfortable with a `conda` based solution, but OK, I will merge this.

File: .github/workflows/wheels.yml [hunk 2]
```diff
@@ -103,12 +115,14 @@ jobs:
...
       matrix:
-        os: [ 'ubuntu-latest', 'windows-latest' ]
+        os: [ 'ubuntu-latest', 'windows-latest', 'macos-latest' ]
         python-version: ['3.7', '3.8', '3.9','3.10']
         arch: ['x64', 'x86']
         exclude:
         - os: 'ubuntu-latest'
           arch: 'x86'
+        - os: 'macos-latest'
+          arch: 'x86'
```
Comment (avalentino) [inline .github/workflows/wheels.yml:126]: Just realised that we are not testing aarch64 (linux), this is not related to this specific PR anyways.

Comment (xmatthias) [reply .github/workflows/wheels.yml:126]: we are not and we really cannot realistically.

github is not offering ARM based runners at the moment, and running tests with emulation (qemu + docker) is pretty slow (30+ minutes to run the "light" test-suite when i last ran that).

File: .github/workflows/wheels.yml (hunk 1 above)
Comment (xmatthias) [reply .github/workflows/wheels.yml:46]: i hope there is no misunderstanding here - you should not feel forced to merge PR's you're not comfortable with.

If there is a reason you think using brew is worse than using conda let me know, it's a PR for a reason (allowing discussions, ...) :)

**Comment by avalentino** [comment]
> I'm also working on a "build hdf5 from source" PR - which will build hdf5 from source on linux/macos Systems - but the brew dependency will probably remain there for some "basic" libraries we don't really want to build.
>
> I'm not sure if it would be better to use the builtin / vendored blosc (I'm not sure on reasons for vendoring that library - but discovered that it'll be built as part of the wheel building process if the library is not there).

Blosc was vendored mostly because it was not available via standard packages not even on linux distributions some times ago.

Today it would be probably not necessary to vendor it but I like your proposal to have a zero-dependency procedure the builds everything for the scratch. Having a static linking would be even better, probably, from a wheel user perspective.

File: .github/workflows/wheels.yml (hunk 1 above)
Comment (avalentino) [reply .github/workflows/wheels.yml:46]: No it is just a personal feeling.
I know conda better than brew so I would prefer to go with conda but this is irrelevant for this PR IMHO.

File: .github/workflows/wheels.yml (hunk 2 above)
Comment (avalentino) [reply .github/workflows/wheels.yml:126]: > running tests with emulation (qemu + docker) is pretty slow (30+ minutes to run the "light" test-suite when i last ran that)

OK you convinced me
This is something that can be implemented in the future when proper runners will be available.
I will file in issue for that as a reminder.
